Core.UnitTests/TypeAssembly/Implementation/TypeAssemblerTest.cs

```diff
@@ -153,90 +152,116 @@ public void AssembleType ()
+                proxyTypeAssemblyContextPassedToParticipant1 = proxyTypeAssemblyContextArg;
+                Assert.That (proxyTypeAssemblyContextPassedToParticipant1.ParticipantConfigurationID, Is.EqualTo (participantConfigurationID));
+                Assert.That (proxyTypeAssemblyContextPassedToParticipant1.ParticipantState, Is.SameAs (_participantStateMock.Object));
+                Assert.That (proxyTypeAssemblyContextPassedToParticipant1.RequestedType, Is.SameAs (_requestedType));
+                Assert.That (proxyTypeAssemblyContextPassedToParticipant1.ProxyType, Is.SameAs (proxyType));
+
+                proxyTypeAssemblyContextPassedToParticipant1.CreateAdditionalType (additionalTypeID, "AdditionalType", null, 0, typeof (int));
+
+                proxyTypeAssemblyContextPassedToParticipant1.GenerationCompleted += ctx =>
                 {
-                  Assert.That (generationCompletedEventRaised, Is.False);
+                  Assert.That (ctx.GetGeneratedType (proxyType), Is.SameAs (fakeGeneratedType));
+                  generationCompletedEventRaised = true;
+                };
+              });
+      var additionalType = MutableTypeObjectMother.Create();
+      mutableTypeFactoryMock
+          .InSequence (sequence)
+          .Setup (mock => mock.CreateType ("AdditionalType", null, 0, typeof (int), null)).Returns (additionalType);
+
+      assembledTypeIdentifierProviderMock
+          .InSequence (sequence)
+          .Setup (mock => mock.GetPart (It.Is<AssembledTypeID> (id => id.Equals (typeID)), participantMock2.Object))
+          .Returns ((object) null);
+      participantMock2
+          .InSequence (sequence)
+          .Setup (mock => mock.Participate (null, It.IsAny<IProxyTypeAssemblyContext>()))
+          .Callback (
+              (object _, IProxyTypeAssemblyContext proxyTypeAssemblyContext) =>
+              {
+                Assert.That (proxyTypeAssemblyContext, Is.EqualTo (proxyTypeAssemblyContextPassedToParticipant1));
+              });
+
+      typeModificationTrackerMock
+          .InSequence (sequence)
+          .Setup (mock => mock.IsModified()).Returns (true);
+
+      assembledTypeIdentifierProviderMock
+          .InSequence (sequence)
+          .Setup (mock => mock.AddTypeID (proxyType, It.Is<AssembledTypeID> (id => id.Equals (typeID))));
+      complexSerializationEnablerMock
+          .InSequence (sequence)
+          .Setup (mock => mock.MakeSerializable (proxyType, participantConfigurationID, assembledTypeIdentifierProviderMock.Object, It.Is<AssembledTypeID> (id => id.Equals (typeID))));
+
+      codeGeneratorMock
+          .InSequence (sequence)
+          .Setup (mock => mock.GenerateTypes (It.Is<IEnumerable<MutableType>> (mutableTypes => mutableTypes.SequenceEqual (new[] { additionalType, proxyType }))))
+          .Returns (new[] { new KeyValuePair<MutableType, Type> (proxyType, fakeGeneratedType), new KeyValuePair<MutableType, Type> (additionalType, fakeGeneratedAdditionalType) })
```

Comment:
keep original linewrap?

Reply:
In the other files you said that we can do the setup in one line to make it occupy less space, I can reintroduce the linewrap if you like, though

Reply:
Don't remember it exactly. here, we originally had a line break for the return and the line is too long anyway so changing to a single line feels wrong to me.